src/main/java/htsjdk/samtools/cram/compression/rans/RANSNx16.java

```diff
@@ -0,0 +1,152 @@
+package htsjdk.samtools.cram.compression.rans;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class RANSNx16 extends RANSNew<RANSNx16Params>{
+
+    private static final ByteBuffer EMPTY_BUFFER = ByteBuffer.allocate(0);
+    private static final int MINIMUM__ORDER_1_SIZE = 4;
+
+    public ByteBuffer compress(final ByteBuffer inBuffer, final RANSNx16Params params) {
+        if (inBuffer.remaining() == 0) {
+            return EMPTY_BUFFER;
+        }
+        final ByteBuffer outBuffer = allocateOutputBuffer(inBuffer.remaining());
+        final int format = params.getFormat();
+        outBuffer.put((byte) (format)); // one byte for format
+        final RANSParams.ORDER order = params.getOrder(); // Order-0 or Order-1 entropy coding
+        final int x32 = params.getX32(); // Interleave N = 32 rANS states (else N = 4)
+        final int stripe = params.getStripe(); //multiway interleaving of byte streams
+        final int nosz = params.getNosz(); // original size is not recorded
+        final int cat = params.getCAT(); // Data is uncompressed
+        final int rle = params.getRLE(); // Run length encoding, with runs and literals encoded separately
+        final int pack = params.getPack(); // Pack 2, 4, 8 or infinite symbols per byte
+
+        // TODO: add methods to handle various flags
+
+        // N-way interleaving
+        final int Nway = (x32 != 0) ? 32 : 4;
+
+        //stripe size
+        final int N = format>>8;
+
+        if (nosz == 0) {
+            int insize = inBuffer.remaining();
+            Utils.writeUint7(insize,outBuffer);
+        }
+        initializeRANSCoder();
+        if (cat != 0) {
+            outBuffer.put(inBuffer);
+            return outBuffer;
+        }
+
+        if (inBuffer.remaining() < MINIMUM__ORDER_1_SIZE) {
+            // TODO: check if this still applies for Nx16 or if there is a different limit
+            // ORDER-1 encoding of less than 4 bytes is not permitted, so just use ORDER-0
+            return compressOrder0WayN(inBuffer, Nway, outBuffer);
+        }
+
+        switch (order) {
+            case ZERO:
+                return compressOrder0WayN(inBuffer, Nway, outBuffer);
+//            case ONE:
+//                return compressOrder1WayN(inBuffer, Nway, outBuffer);
+            default:
+                throw new RuntimeException("Unknown rANS order: " + order);
+        }
+    }
+
+    public ByteBuffer uncompress(final ByteBuffer inBuffer, int n_out) {
```

Comment:
Also, if this argument (`n_out`) is the only `uncompress` arg, I think it would be fine unify the `uncompress` method across 4x8 and Nx16 so they have the same signature, and just have the 4x8 implementation ignore this value. The the `uncompress` method can be moved in to the `RANSNew` interface.